pinot-core/src/main/java/org/apache/pinot/core/query/reduce/DistinctDataTableReducer.java

```diff
@@ -108,7 +108,7 @@ public void reduceAndSetResults(String tableName, DataSchema dataSchema,
       if (mergedIntermediateResult == null) {
         mergedIntermediateResult = intermediateResultToMerge;
       } else {
-        _aggregationFunction.merge(mergedIntermediateResult, intermediateResultToMerge);
+        mergedIntermediateResult = _aggregationFunction.merge(mergedIntermediateResult, intermediateResultToMerge);
```

Comment:
IIUC, this is needed because for the case of empty, it may return a new object as opposed to merging into existing ones?

Reply:
Yes, and we should not make the assumption that the second result is merged into the first (for Double intermediateResult, that is not even possible)